espressif/vscode-esp-idf-extension
# Enhance: Enable usage of white spaces in project's path

## Description

Enable usage of white spaces in project's path.
https://jira.espressif.com:8443/browse/VSC-808

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output
1. Make sure all the above have whitespaces in their paths:

- ESP-IDF folder
- ESP-IDF Tools folder
- Your project
>Note: For ESP-IDF folder and ESP-IDF Tools folder it means you will need to make a new installation for them.
2. Create a project that has a white space in it. Ex: "My Project"
3. Build, Flash, Monitor it.
>Note: Needs to be tested for all variations of: **flashtype method** (dfu, jtag, uart), **Default Terminal in VsCode**:
Windows: Command Prompt and Powershell
Mac: zsh/bash
Linux: bash
 (open command palette in VS Code [CTRL + SHIFT +P] and use the command: `Terminal: Select Default Profile` 
6. There should be no output error regarding the paths. Such as 'invalid path' or 'invalid argument' etc.

- Expected behaviour:
With these changes, user should be able to use the VsCode extension and it's features on projects that have white spaces in their path or in the ESP-IDF and ESP-IDF Tools installation paths

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Tested building, flashing and monitoring on all variations of Terminals (Windows: Powershell, Mac: zsh/bash, Linux: bash) and flashing methods (DFU, JTAG, UART).

**Test Configuration**:
* ESP-IDF Version: 5.1
* OS (Windows,Linux and macOS): Windows, Linux, macOS

## Checklist
